File: valid-palindrome-ii/valid-palindrome-ii.go
func validPalindrome(s string) bool {
    var isPalindrome func(int, int, bool) bool
    isPalindrome = func(start, end int, skipAvailable bool) bool {
        for i, j := start, end; i <= j; i, j = i+1, j-1 {
            if s[i] != s[j] {
                if skipAvailable {
                    return isPalindrome(i+1, j, false) || isPalindrome(i, j-1, false)
                }
                return false
            }
        }
        return true
    }
    
    return isPalindrome(0, len(s)-1, true)
}

File: valid-palindrome-ii/valid-palindrome-ii.py
class Solution:
    def helper(self,s,start,end):
        while start<=end:
            if s[start]!=s[end]:
                return False
            start+=1
            end-=1
        return True
    def validPalindrome(self, s: str) -> bool:
        low=0
        high=len(s)-1
        while low<=high:
            if s[low]!=s[high]:
                return (self.helper(s,low+1,high)) or (self.helper(s,low,high-1))
            low+=1
            high-=1
        return True